server/src/server.ts

```diff
@@ -205,20 +205,49 @@ let sendCompilationFinishedMessage = () => {
   send(notification);
 };
 
+let debug = false;
+
+let syncProjectConfigCache = (rootPath: string) => {
+  try {
+    if (debug) console.log("syncing project config cache for " + rootPath);
+    utils.runAnalysisAfterSanityCheck(rootPath, ["cache-project", rootPath]);
+    if (debug) console.log("OK - synced project config cache for " + rootPath);
+  } catch (e) {
+    if (debug) console.error(e);
+  }
+};
+
+let deleteProjectConfigCache = (rootPath: string) => {
+  try {
+    if (debug) console.log("deleting project config cache for " + rootPath);
+    utils.runAnalysisAfterSanityCheck(rootPath, ["cache-delete", rootPath]);
+    if (debug) console.log("OK - deleted project config cache for " + rootPath);
+  } catch (e) {
+    if (debug) console.error(e);
+  }
+};
+
 let compilerLogsWatcher = chokidar
   .watch([], {
     awaitWriteFinish: {
       stabilityThreshold: 1,
     },
   })
-  .on("all", (_e, _changedPath) => {
-    sendUpdatedDiagnostics();
-    sendCompilationFinishedMessage();
-    if (config.extensionConfiguration.inlayHints?.enable === true) {
-      sendInlayHintsRefresh();
-    }
-    if (config.extensionConfiguration.codeLens === true) {
-      sendCodeLensRefresh();
+  .on("all", (_e, changedPath) => {
+    if (changedPath.includes("build.ninja")) {
+      let projectRoot = utils.findProjectRootOfFile(changedPath);
+      if (projectRoot != null) {
+        syncProjectConfigCache(projectRoot);
```

Comment:
maybe make this conditional on the setting, so potential issues of writing this file would be contained to having the setting on.

Reply:
Definitely!

Reply:
Although the watch wouldn't be set up unless the setting is on.

Reply:
Fair point :)